
        <!DOCTYPE html>
        <html>
            <head>
                <meta http-equiv="Content-Type" content="text/html; charset=gb2312">
                <title>SNP-based Heritability - Jimmy Z. Hu's Blog</title>
                <style rel="stylesheet" type="text/css">
                    body {font-family: Georgia; font-size: 17px;}
h2 {margin-top: 40px; border-bottom: 1px solid #999; text-transform: uppercase;}
p {text-align: left; line-height: 1.4em;}
pre {
    background: #f4f4f4; border: 1px solid #ddd; border-left: 3px solid #9f9f9f; 
    color: #666; page-break-inside: avoid; font-family: monospace; font-size: 15px; 
    line-height: 1.1; max-width: 100%; overflow: auto; padding: 0.5em 0.5em; 
    display: block; word-wrap: break-word;
}
li {line-height: 1.25em;}
a:link {text-decoration: none; color: blue;}
a:visited {text-decoration: none; color: blue;}
.container {width: 750px; margin-left: auto; margin-right: auto; padding: 50px 0 50px 0;}
.title {text-align: left;}
.katex {font-size: 1em !important;} 
table {width: 100%%;}
table, th, td {border: 1px solid black;}

                </style>
                <link rel="stylesheet" href="https://cdn.jsdelivr.net/npm/katex@0.12.0/dist/katex.min.css" integrity="sha384-AfEj0r4/OFrOo5t7NnNe46zW/tFgW6x/bCJG8FqQCEo3+Aro6EYUG4+cU+KJWu/X" crossorigin="anonymous">
                <script defer src="https://cdn.jsdelivr.net/npm/katex@0.12.0/dist/katex.min.js" integrity="sha384-g7c+Jr9ZivxKLnZTDUhnkOnsh30B4H0rpLUpJ4jAIKs4fnJI+sEnkvrMWph2EDg4" crossorigin="anonymous"></script>
                <script defer src="https://cdn.jsdelivr.net/npm/katex@0.12.0/dist/contrib/auto-render.min.js" integrity="sha384-mll67QQFJfxn0IYznZYonOWZ644AWYC+Pt2cHqMaRhXVrursRwvLnLaebdGIlYNa" crossorigin="anonymous"></script>
                <script>
                    document.addEventListener("DOMContentLoaded", function() {
                        renderMathInElement(document.body, {
                            delimiters: [
                              {left: "$$", right: "$$", display: true},
                              {left: "$", right: "$", display: false}
                            ] 
                        });
                    });
                </script>
            </head>
            <body>
                <div class="container">
                    <h1 class="title">SNP-based Heritability</h1>
                    2020-05-31
                    <span style="float:right; _position:relative;">
                        <a href="/home">HOME</a>
                        &ensp;
                        <a href="/notes">NOTES</a> &ensp; <a href="../">RELATED</a>
                    </span><br/>
                    <h2>URLs</h2>
<ul>
<li>GCTA, http://cnsgenomics.com/software/gcta/</li>
<li>PCGC, https://www.hsph.harvard.edu/alkes-price/software/</li>
<li>LDSC, https://github.com/bulik/ldsc</li>
</ul>
<h2><strong>Yang et al. 2017</strong></h2>
<ul>
<li>Yang, J., Zeng, J., Goddard, M. E., Wray, N. R., &amp; Visscher, P. M. (2017). Concepts, estimation and interpretation of SNP-based heritability. Nature genetics, 49(9), 1304.</li>
</ul>
<p>Estimation of the variance explained by all SNPs used in a population based GWAS was initially motivated by the 'missing heritability' problem: <strong>the estimated variance explained by genome-wide significant (GWS) SNPs discovered in GWAS (denoted $\pmb{\hat{h}^2_{\mathrm{GWS}}}$) was only a fraction of the estimated heritability ($\pmb{\hat{h}^2}$)</strong> from family or twin studies, where $\hat{h}^2_{\mathrm{GWS}}$ was estimated in a multi-SNP model to account for linkage disequilibrium (LD) among SNPs and in an independent sample to avoid overestimation due to winner's curse.</p>
<p>SNP-based heritability (or $h^2_{\mathrm{SNP}}$) was initially defined as the proportion of phenotypic variance explained by all SNPs on a genotyping array and is therefore dependent of the number of SNPs on a SNP array.</p>
<p>For complex traits like height, <strong>there are likely a large number of common variants with effect sizes too small to pass the stringent GWS threshold ($\pmb{p &lt; 5 × 10^{-8}}$) in GWAS</strong>, even with sample sizes that were considered large at that time ($n$ = 1,000 to 10,000 samples before 2010), consistent with a model of polygenic inheritance.</p>
<p>The approach of Yang et al. (2010) was subsequently termed genomic relatedness matrix (GRM) restricted maximum likelihood (GREML) and implemented in the <strong>GCTA software tool</strong>. The GREML estimate directly quantifies the proportion of phenotypic variance explained by all SNPs used in GWAS and therefore provides the upper limit of $\hat{h}^2_{\mathrm{GWS}}$ given the same experimental design. $h^2_{\mathrm{SNP}}$ is by definition smaller than $h^2$ because not all causal variants, in particular those with low frequency, can be perfectly tagged by SNPs used in GWAS.</p>
<p>A sparse SNP array that does not cover common variation in the genome well is less likely to lead to the discovery of trait-associated variants (even with a large sample size), and fitting those SNPs together in a GREML analysis will result in a smaller proportion of phenotypic variance explained than with a denser SNP array. Because the maximum possible LD correlation between two genetic variants declines as their difference in MAF increases, genetic variation at rare variants (MAF &lt; 0.01) is unlikely to be well tagged by common SNPs on genotyping arrays. If causal variants are located in genomic regions with a different LD property from the rest of the genome, this can lead to bias in  $h^2_{\mathrm{SNP}}$.</p>
<center><img src='box1.png' width=80%></center>
<p>GREML is based on a random-effect model (Box 1). If the number of SNPs ($m$) is smaller than the sample size ($n$), this model is similar to a linear regression analysis (fixed-effect model). In GWAS, $m$ is normally larger than $n$, in which case there is no unique solution to the fixed-effect model. In a random-effect model, there is an additional assumption that the joint SNP effects $u = \left\{ u_1, u2, \dots, u_m \right\}$ follow a normal distribution with mean 0 and variance $\sigma_u^2$ (see Box 1 for notations) so that the model parameters are estimable even when $m$ is larger than $n$, where $\sigma_u^2$ is interpreted as per-SNP genetic variance when all SNPs are fitted jointly, hence accounting for LD.</p>
<p>In addition, there is often a question about whether the SNPs included in GREML analysis need to be pruned for LD. As discussed above, GREML accounts for LD so that LD pruning is not necessary.</p>
<p>The unbiasedness of GREML in estimating $h^2$ using WGS data depends on the ratio of $r_{MQ}^2$ (mean LD $r^2$ between causal and non-causal variants) to $r_{MM}^2$ (mean LD $r^2$ between non-causal variants). Note that, because $r^2$ is a function of MAF, a difference in MAF spectrum between causal and non-causal variants will lead to a difference in LD.</p>
<p>One solution is to stratify SNPs by MAF <strong>(MAF-stratified GREML, GREML-MS)</strong></p>
<p>A more general approach, <strong>GREML-LDMS</strong>,  is to not rely on a specific model of the interplay between allele frequency, effect size and LD, but instead stratify SNPs by MAF and LD jointly and estimate genetic variance with MAF–LD subsets.</p>
<p>In GCTA-GREML, we standardize the SNP genotypes and assume that the effect size per standardized genotype ($u_i$) follows a normal distribution. GREML-MS allows the data to reveal the relationship between variance explained and MAF. The REML algorithm is computationally intensive. Alternatively, $\sigma^2_{g(\mathrm{SNP})}$ can be estimated from Haseman–Elston (HE) regression, that is, $y_iy_j=b_0+b_1A_{ij}+e_{ij}$, where $b_1=\sigma^2_{g(\mathrm{SNP})}$. HE regression is computationally much more efficient but slightly less powerful than REML, as the SE of $\hat{h}^2_{\mathrm{SNP}}$ from HE regression is larger than that from REML. The small difference in SE between the methods might not be important when the sample size becomes very large.</p>
<h2><strong>Evans et al., 2018</strong></h2>
<ul>
<li>Evans, L. M., Tahmasbi, R., Vrieze, S. I., Abecasis, G. R., Das, S., Gazal, S., ... &amp; Yang, J. (2018). Comparison of methods that use whole genome data to estimate the heritability and genetic architecture of complex traits. Nature genetics, 50(5), 737-745.</li>
</ul>
<p>As SNPs in these analyses begin to more accurately reflect the density and frequency distributions of CVs, $\hat{h}^2_{\mathrm{SNP}}$ should approach total $h_2$ , making it important to understand the factors that can bias $\hat{h}^2_{\mathrm{SNP}}$. Moreover, the proliferation of methods has led to discrepancies in estimates.</p>
<center><img src='table1.png' width=100%></center>
<center><img src='fig1.png' width=100%></center>
<p>LD is central to the performance of all the methods compared here, particularly the LD among SNPs used to create the GRM and that between CVs and SNPs. Single-component models, such as GREML-SC and LDAK-SC, are highly sensitive to assumptions, especially when rare imputed or WGS SNPs are used to create the GRM. Alternatively, multicomponent models that partition $\hat{h}^2_{\mathrm{SNP}}$ across multiple LD and MAF bins provide the most robust estimates across the majority of contexts explored here, while simultaneously providing insight into the allelic spectra of complex traits. However, they are more computationally intensive and have higher standard errors than single-component models, and they require larger datasets to achieve reliable estimates. Nevertheless, such data are now at hand, and if the goal is to obtain the least biased estimates of $h^2$ or to estimate allelic spectra, it is recommended to use multicomponent GREML models.</p>
<p>First, <strong>quality control of genetic data is crucial</strong>, particularly for case–control and/or multiple-cohort datasets. <strong>Covariates (ancestry principal components, cohorts, plates, etc.) that might be confounded with genetic similarity should be included as fixed effects in GREML models and in the GWAS models for LDSC</strong>. Related individuals may share common environmental and nonadditive genetic effects, upwardly biasing estimates of $h^2$; using unrelated individuals should provide estimates not inflated by such factors.</p>
<p>Second, <strong>the model and data type used in the analysis strongly influence estimates</strong>. Single-component methods such as GREML-SC and LDAK-SC are highly sensitive to model misspecification. Moreover, they are also sensitive to the effects of stratification. We recommend these approaches only when sample sizes are small (for example, $n$&lt;30,000) and homogeneous. Multicomponent approaches with WGS or imputed SNPs provide the most accurate estimates of $h^2$  and allelic spectra across a range of genetic architectures and stratification levels. When using imputed data, SNPs should be imputed using the largest and most diverse reference panel possible in order to more reliably capture the effects of rare CVs. However, more GRMs lead to larger standard errors, necessitating larger sample sizes ($n$&gt;30,000). <strong>Of the multicomponent approaches, GREML-LDMS-I, which we introduce here and which bins SNPs by MAF and individual LD levels, appears to perform the best.</strong></p>
<h2><strong>Hou et al. 2019</strong></h2>
<ul>
<li>Hou, K., Burch, K. S., Majumdar, A., Shi, H., Mancuso, N., Wu, Y., ... &amp; Pasaniuc, B. (2019). Accurate estimation of SNP-heritability from biobank-scale data irrespective of genetic architecture. Nature genetics, 1.</li>
</ul>
<p>All REML-based methods are biased when their model assumptions are violated, and multi-component REML methods that stratify SNPs by MAF and LD score (GREML-LDMS-I) are more accurate than single-component REML methods. The performance of the generalized random effects (GRE) estimator is similar to that of GREML-LDMS-I, thereby confirming that SNP-heritability can be accurately estimated without stratifying SNPs or specifying a heritability model</p>
<center><img src='fig3.png' width=100%></center>
<p>First, the utility of $\hat{h}^2_{\mathrm{GRE}}$ critically depends on the ratio between the number of SNPs ($M$) and the number of individuals ($N$)—<strong>as $M/N$ increases, the eigenstructure of the in-sample LD matrix becomes increasingly distorted</strong> (larger eigenvalues are overestimated; smaller eigenvalues are underestimated). <strong>We mitigate this by assuming that chromosomes are approximately independent</strong>; as long as N exceeds the number of array SNPs per chromosome, $\hat{h}^2_{\mathrm{GRE}}$ provides meaningful estimates of SNP-heritability</p>
<p>Second, the theoretical guarantees of $\hat{h}^2_{\mathrm{GRE}}$ rely on the assumption that OLS association statistics and LD are estimated from the same genotypes. While summary statistics have been made publicly available for hundreds of large-scale genome-wide association studies (GWAS), in-sample LD is usually unavailable for these studies since most are meta-analyses. <strong>In addition, summary statistics are often computed using linear mixed models to control for confounding, and previous works have noted that the LD computation must be adjusted to accommodate mixed model association statistics</strong>. Thus, the sensitivity of $\hat{h}^2_{\mathrm{GRE}}$ to reference panel LD (with or without regularized LD estimation) and/or mixed model association statistics remains unclear</p>
<p>Third, $\hat{h}^2_{\mathrm{GRE}}$ does not correct for population structure/stratification. In real data, we mitigate this by considering only unrelated individuals (&gt; third-degree relatives) and including age, sex and the top 20 PCs as covariates when computing association statistics. While recent work has found evidence of assortative mating for some traits in the UK Biobank (for example, height), our estimates are robust to different relatedness thresholds, suggesting that <strong>adjusting for the top 20 PCs sufficiently controls for population stratification</strong>.</p>

                    <p style="border-top: 1px solid #dbdbdb; margin-top: 40px;"></p>
                    <div id="valine-thread"></div>
                    <script src="//cdn1.lncld.net/static/js/3.0.4/av-min.js"></script>
                    <script src='/js/Valine.min.js'></script>
                    <script>
                        new Valine({
                            el: '#valine-thread',
                            appId: 'DSTtKETb5UeyXIjoPTpRnC8Y-gzGzoHsz',
                            appKey: 'csHaHWqxD2Ujv84O7jaJWOSc',
                            verify: false,
                            avatar: 'robohash',
                            placeholder: '',
                            meta: ['nick', 'mail'],
                            requiredFields: ['nick'],
                            visitor: true,
                            lang: 'en'
                        })
                    </script>
                  
                    <footer>
                        <p style="padding: 5px 0 0 0; text-align: center; font-size: .9rem; border-top: 1px solid #dbdbdb;margin-top: 40px;">&#169; 2021 Jimmy Z. Hu<br>This website is licensed under <a href="https://creativecommons.org/licenses/by-nc-sa/4.0/">CC BY-NC-SA 4.0.</a></p>
                    </footer>
                </div>
            </body>
        </html>
    