
        <!DOCTYPE html>
        <html>
            <head>
                <meta http-equiv="Content-Type" content="text/html; charset=gb2312">
                <title>GWAS Batch Processing - Jimmy Z. Hu's Blog</title>
                <style rel="stylesheet" type="text/css">
                    body {font-family: Georgia; font-size: 17px;}
h2 {margin-top: 40px; border-bottom: 1px solid #999; text-transform: uppercase;}
p {text-align: left; line-height: 1.4em;}
pre {
    background: #f4f4f4; border: 1px solid #ddd; border-left: 3px solid #9f9f9f; 
    color: #666; page-break-inside: avoid; font-family: monospace; font-size: 15px; 
    line-height: 1.1; max-width: 100%; overflow: auto; padding: 0.5em 0.5em; 
    display: block; word-wrap: break-word;
}
li {line-height: 1.25em;}
a:link {text-decoration: none; color: blue;}
a:visited {text-decoration: none; color: blue;}
.container {width: 750px; margin-left: auto; margin-right: auto; padding: 50px 0 50px 0;}
.title {text-align: left;}
.katex {font-size: 1em !important;} 
table {width: 100%%;}
table, th, td {border: 1px solid black;}

                </style>
                <link rel="stylesheet" href="https://cdn.jsdelivr.net/npm/katex@0.12.0/dist/katex.min.css" integrity="sha384-AfEj0r4/OFrOo5t7NnNe46zW/tFgW6x/bCJG8FqQCEo3+Aro6EYUG4+cU+KJWu/X" crossorigin="anonymous">
                <script defer src="https://cdn.jsdelivr.net/npm/katex@0.12.0/dist/katex.min.js" integrity="sha384-g7c+Jr9ZivxKLnZTDUhnkOnsh30B4H0rpLUpJ4jAIKs4fnJI+sEnkvrMWph2EDg4" crossorigin="anonymous"></script>
                <script defer src="https://cdn.jsdelivr.net/npm/katex@0.12.0/dist/contrib/auto-render.min.js" integrity="sha384-mll67QQFJfxn0IYznZYonOWZ644AWYC+Pt2cHqMaRhXVrursRwvLnLaebdGIlYNa" crossorigin="anonymous"></script>
                <script>
                    document.addEventListener("DOMContentLoaded", function() {
                        renderMathInElement(document.body, {
                            delimiters: [
                              {left: "$$", right: "$$", display: true},
                              {left: "$", right: "$", display: false}
                            ] 
                        });
                    });
                </script>
            </head>
            <body>
                <div class="container">
                    <h1 class="title">GWAS Batch Processing</h1>
                    2020-04-12
                    <span style="float:right; _position:relative;">
                        <a href="/home">HOME</a>
                        &ensp;
                        <a href="/notes">NOTES</a> &ensp; <a href="../">RELATED</a>
                    </span><br/>
                    <h3>Deal with Data Stored in Different Filesets</h3>
<p>If SNP data is stored in different files, merging all filesets can be extremely helpful in conducting sex, missingness, and heterozygosity checking. Commands are shown below:</p>
<pre><code>--merge [.ped filename] [.map filename]<br>--merge [text fileset prefix]<br>--bmerge [.bed filename] [.bim filename] [.fam filename]<br>--bmerge [binary fileset prefix]<br>--merge-mode [mode number]<br>--merge-equal-pos</code></pre>
<p><a href="https://www.cog-genomics.org/plink/1.9/data#merge">Plink documentation</a> gives more details about how to merge file.</p>
<h3>Solve Memory Error in Checking Duplication or Relatedness</h3>
<p>Normally, generating pair-wise IBS for all pairs of individuals over the whole dataset requires an extremely large memory space. However, most of PCs cannot afford such huge computational cost. It would be wise to break this large task down into smaller pieces.<details><summary><i>Plink Results</i></summary><pre style=white-space:pre-wrap;word-wrap:break-word;><code>PLINK v1.90b6.16 64-bit (19 Feb 2020)<br>Options in effect:<br>  --bfile data<br>  --extract data.prune.in<br>  --genome<br>  --out data<br><br>Random number seed: 1587131431<br>8066 MB RAM detected; reserving 4033 MB for main workspace.<br>805426 variants loaded from .bim file.<br>218545 people (99683 males, 118862 females) loaded from .fam.<br>--extract: 404542 variants remaining.<br>Using up to 8 threads (change this with --threads).<br>Before main variant filters, 218545 founders and 0 nonfounders present.<br>Calculating allele frequencies... done.<br>Total genotyping rate is 0.967801.<br>404542 variants and 218545 people pass filters and QC.<br>Note: No phenotypes present.<br><br>Error: Out of memory.  The --memory flag may be helpful.<br>Failed allocation size: 95523396992
</code></pre></p>
</details>
<p><a href="https://www.ncbi.nlm.nih.gov/pmc/articles/PMC1950838/">Purcell et al., 2007</a> introduces how Plink applies method of moments to estimate IBD, which can be derived by using Hardy–Weinberg equations to calculate the conditional probabilities of observable identity by state (IBS) patterns given IBD states and applying Bayes formula. In this method, major and minor sample allele frequencies are scaled by a constant to correct for finite sample size and then used to approximate population allele frequencies (<a href="https://www.ncbi.nlm.nih.gov/pmc/articles/PMC4001853/">Morrison, 2013</a>). A practical way for estimating IBD using computer can be found in Plink source code (<a href="https://github.com/chrchang/plink-ng/blob/master/1.9/plink_calc.c">plink_calc.c</a> and <a href="https://github.com/GELOG/plink/blob/master/plink-1.07-src/genome.cpp">genome.cpp</a>).</p>
<p>Based on theories behind IBD estimation in Plink, we can initially divide the estimation process in to two parts: one is to approximate population allele frequencies, another is to estimate IBD for every pair of individuals. Generally, Plink will automatically calculate sample allele frequencies based the input dataset, and treat them as population allele frequencies, which causes a huge bias in estimating population allele frequencies. So, here we call <strong>--genome</strong> with an additional option <strong>--read-freq</strong> to force Plink to use the .frq we have previously created instead of letting Plink calculate the sample allele frequencies of input dataset.</p>
<p>For estimating IBD, we use <strong>--parallel</strong> and <strong>--memory</strong> options to break the huge task into several smaller part. More detials can be found on <a href="https://www.cog-genomics.org/plink/1.9/parallel">Plink's homepage</a></p>
<pre><code>--parallel [1-based current job index] [total job pieces]</code></pre>
<p>It will take some time to do experiment and find the best parameters for <strong>--parallel</strong>. When the task is extremely huge, we need to type bunches of commands in shell. So I used Python (<a href="https://github.com/ZijingHu/GWAS/blob/master/distributed-genome.py">code</a>) to generate a .sh file to include all of these commands.</p>

                    <p style="border-top: 1px solid #dbdbdb; margin-top: 40px;"></p>
                    <div id="valine-thread"></div>
                    <script src="//cdn1.lncld.net/static/js/3.0.4/av-min.js"></script>
                    <script src='/js/Valine.min.js'></script>
                    <script>
                        new Valine({
                            el: '#valine-thread',
                            appId: 'DSTtKETb5UeyXIjoPTpRnC8Y-gzGzoHsz',
                            appKey: 'csHaHWqxD2Ujv84O7jaJWOSc',
                            verify: false,
                            avatar: 'robohash',
                            placeholder: '',
                            meta: ['nick', 'mail'],
                            requiredFields: ['nick'],
                            visitor: true,
                            lang: 'en'
                        })
                    </script>
                  
                    <footer>
                        <p style="padding: 5px 0 0 0; text-align: center; font-size: .9rem; border-top: 1px solid #dbdbdb;margin-top: 40px;">&#169; 2021 Jimmy Z. Hu<br>This website is licensed under <a href="https://creativecommons.org/licenses/by-nc-sa/4.0/">CC BY-NC-SA 4.0.</a></p>
                    </footer>
                </div>
            </body>
        </html>
    